### PR TITLE
Update/5.1.0

### PR DESCRIPTION
**Pull request by ryanskeith**
**Destination channel:** main

### Links

- [PKG-3663](https://anaconda.atlassian.net/browse/PKG-3663) 
- [Upstream repository](https://www.gaia-gis.it/fossil/libspatialite/home)

### Explanation of changes:

- update version and sha
- updated build scripts to target geos < 3.10, no rttopo. rttopo is GPL
- updated patches - especially, a number of tests call a convenience init function which rely on rttopo.

[PKG-3663]: https://anaconda.atlassian.net/browse/PKG-3663?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,32 +32,33 @@ requirements:
...
     - proj
     - sqlite
     - zlib
+    - minizip
 
 about:
   home: https://www.gaia-gis.it/fossil/libspatialite/index
-  license: LGPL-2.1-or-later
-  license_family: LGPL
+  license: MPL-1.1 OR GPL-2.0 OR LGPL-2.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:57]:
```suggestion
  license: MPL-1.1 OR GPL-2.0-or-later OR LGPL-2.1-or-later
```

File: abs.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+aggregate_branch: 3.12
```
Comment (skupr-anaconda) [inline abs.yaml:1]: As it's not a Python package the abs.yaml file can be removed

Comment (cbouss) [reply abs.yaml:1]: The 3.12 branch has an updated pin for a more recent version of proj.

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by SC426 — APPROVED** [review]

File: recipe/patches/0001-win_fix_paths.patch [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+==================================================================
+--- src/headers/spatialite/gaiaconfig-msvc.h
++++ src/headers/spatialite/gaiaconfig-msvc.h
+@@ -1,19 +1,19 @@
+ /* ./src/headers/spatialite/gaiaconfig-msvc.h.  Generated from gaiaconfig-msvc.h.in by configure.  */
+ /* ./src/headers/spatialite/gaiaconfig-msvc.h.in - manually maintained */
+ 
+ /* Should be defined in order to enable GCP support. */
+-#define ENABLE_GCP 1
++#undef ENABLE_GCP
```
Comment (JeanChristopheMorinPerso) [inline recipe/patches/0001-win_fix_paths.patch:207]: Should this go in another patch to differentiate our patches from conda-forge patches?

Comment (cbouss) [reply recipe/patches/0001-win_fix_paths.patch:207]: I believe all changes should be in the same patch. conda-forge builds with gpl components. They do not remove rttopo. Do you think 0001-win_fix_paths_and_no_gpl.patch would be better?

Comment (JeanChristopheMorinPerso) [reply recipe/patches/0001-win_fix_paths.patch:207]: https://github.com/conda-forge/libspatialite-feedstock/blob/main/recipe/0001-win_fix_paths.patch isn't GPL related right? So technicall you could move all the GPL stuff into 0003 and rename 0003 `0003-no-gpl.patch` or something like that. This way the conda-forge patch (0002) will stay intact.

How about that?

Comment (cbouss) [reply recipe/patches/0001-win_fix_paths.patch:207]: ok, I can do that

Comment (cbouss) [reply recipe/patches/0001-win_fix_paths.patch]: The cf patch also enables features of geos we don't support. In the end I just renamed the patch.
